transform/app/transform/live_update/data_type_handlers.py

```diff
@@ -0,0 +1,37 @@
+from app.tasks.batch import transform_batch
+from app.tasks.delete_data_by_id import delete_data_by_id
+from app.services.solr.validate import check_document_exists
+from app.transform.schemas.properties.data import ID
+from app.settings import settings
+
+
+def update_data_item(data, data_type, linked_data_type):
+    """
+    Generic function to update data items and delete linked items if necessary.
+
+    Args:
+        data (dict): The data to be updated.
+        data_type (str): The type of the data being updated.
+        linked_data_type (str): The type of the data linked to the main data, which may need deletion.
+
+    Returns:
+        list: A list of Celery task IDs.
+    """
+    tasks = []
+    if check_document_exists(linked_data_type, data[ID]):
+        delete_task = delete_data_by_id.delay(linked_data_type, data, delete=True)
+        tasks.append(delete_task.id)
```

Comment:
please add those tasks as key-value pairs in a dict - key should tell whether it is delete or update task :D